packages/fx-core/tests/core/FxCore.test.ts

```diff
@@ -447,6 +449,65 @@ describe("Core basic APIs", () => {
     }
   });
 
+  it("phantomMigrationV3 return error for invalid project", async () => {
+    const restore = mockedEnv({
+      TEAMSFX_V3: "true",
+    });
+    sandbox.stub(projectMigratorV3, "checkActiveResourcePlugins").resolves(false);
+    try {
+      const core = new FxCore(tools);
+      const appName = await mockV2Project();
+      const inputs: Inputs = {
+        platform: Platform.VSCode,
+        projectPath: path.join(os.tmpdir(), appName),
+        skipUserConfirm: true,
+      };
+      const res = await core.phantomMigrationV3(inputs);
+      assert.isTrue(res.isErr());
```

Comment:
assert which error (NoNeedUpgrade or InvalidProject) it throws, so we can ensure users can get the correct message

Reply:
please also assert error for other test cases

Reply:
Fix it, thanks.